src/Ads/AssetSuggestionsService.php

```diff
@@ -24,16 +26,63 @@ class AssetSuggestionsService implements Service {
 	 * Default maximum marketing images.
 	 */
 	protected const DEFAULT_MAXIMUM_MARKETING_IMAGES = 20;
+	/**
+	 * The subsize key for the square marketing image.
+	 */
+	protected const SQUARE_MARKETING_IMAGE_KEY = 'gla_square_marketing_asset';
+	/**
+	 * The subsize key for the marketing image.
+	 */
+	protected const MARKETING_IMAGE_KEY = 'gla_marketing_asset';
+	/**
+	 * The subsize key for the logo image.
+	 */
+	protected const LOGO_IMAGE_KEY = 'gla_logo_asset';
+	/**
+	 * The minimum and recommended size for a square marketing image. The first value is the width and the second is the height.
+	 */
+	protected const MARKETING_SQUARE_IMAGE_SIZES = [
+		'minimum'     => [ 300, 300 ],
+		'recommended' => [ 1200, 1200 ],
+	];
+	/**
+	 * The minimum and recommended size for a marketing image. The first value is the width and the second is the height.
+	 */
+	protected const MARKETING_IMAGE_SIZES = [
+		'minimum'     => [ 600, 314 ],
+		'recommended' => [ 1200, 628 ],
+	];
+	/**
+	 * The minimum and recommended size for the logo image. The first value is the width and the second is the height.
+	 */
+	protected const LOGO_IMAGE_SIZES = [
+		'minimum'     => [ 128, 128 ],
+		'recommended' => [ 1200, 1200 ],
+	];
+
+	/**
+	 * Image requirements.
+	 *
+	 * @var array
+	 */
+	protected $image_requirements = [];
 
 	/**
 	 * AssetSuggestionsService constructor.
 	 *
-	 * @param WP $wp WP Proxy.
-	 * @param WC $wc WC Proxy.
+	 * @param WP           $wp WP Proxy.
+	 * @param WC           $wc WC Proxy.
+	 * @param ImageUtility $image_utility Image utility.
 	 */
-	public function __construct( WP $wp, WC $wc ) {
-		$this->wp = $wp;
-		$this->wc = $wc;
+	public function __construct( WP $wp, WC $wc, ImageUtility $image_utility ) {
+		$this->wp            = $wp;
+		$this->wc            = $wc;
+		$this->image_utility = $image_utility;
+
+		$this->image_requirements[ self::MARKETING_IMAGE_KEY ]        = self::MARKETING_IMAGE_SIZES;
+		$this->image_requirements[ self::SQUARE_MARKETING_IMAGE_KEY ] = self::MARKETING_SQUARE_IMAGE_SIZES;
+		$this->image_requirements[ self::LOGO_IMAGE_KEY ]             = self::LOGO_IMAGE_SIZES;
```

Comment:
❓  Why is this chunk of code in the constructor and not at the top? As far as I see is not dynamically depending of any constructor argument.